static/js/components/ShareDataForm.jsx

```diff
@@ -0,0 +1,238 @@
+import TextField from "@material-ui/core/TextField";
+
+import { showNotification } from "baselayer/components/Notifications";
+
+import FormValidationError from "./FormValidationError";
+import Plot from './Plot';
+
+import * as photometryActions from "../ducks/photometry";
+import * as spectraActions from "../ducks/spectra";
+import * as sourceActions from "../ducks/source";
+import styles from "./Source.css";
+
+
+const createPhotRow = (id, mjd, mag, magerr, limiting_mag, instrument, filter, groups) => (
+  {
+    id,
+    mjd: Number(mjd).toFixed(3),
+    mag: mag === null ? null : Number(mag).toFixed(4),
+    magerr: magerr === null ? null : Number(magerr).toFixed(4),
+    limiting_mag: Number(limiting_mag).toFixed(4),
+    instrument,
+    filter,
+    groups
+  }
+);
+
+const createSpecRow = (id, instrument, observed, groups) => (
+  {
+    id,
+    instrument,
+    observed,
+    groups
+  }
+);
+
+const photHeadCells = [
+  { name: "id", label: "ID" },
+  { name: "mjd", label: "MJD" },
+  { name: "mag", label: "Mag" },
+  { name: "magerr", label: "Mag Error" },
+  { name: "limiting_mag", label: "Limiting Mag" },
+  { name: "instrument", label: "Instrument" },
+  { name: "filter", label: "Filter" },
+  { name: "groups", label: "Existing Groups" },
+];
+
+const specHeadCells = [
+  { name: "id", label: "ID" },
+  { name: "instrument", label: "Instrument" },
+  { name: "observed", label: "Observed (UTC)" },
+  { name: "groups", label: "Existing Groups" },
+];
+
+const useStyles = makeStyles(() => ({
+  groupSelect: {
+    width: "20rem",
+  },
+}));
+
+const ShareDataForm = ({ route }) => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const [selectedPhotRows, setSelectedPhotRows] = useState([]);
+  const [selectedSpecRows, setSelectedSpecRows] = useState([]);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const groups = useSelector((state) => state.groups);
+  const photometry = useSelector((state) => state.photometry);
+  const spectra = useSelector((state) => state.spectra);
+  const allGroups = groups.all;
+
+  const { handleSubmit, errors, reset, control, getValues } = useForm();
+
+  useEffect(() => {
+    dispatch(photometryActions.fetchSourcePhotometry(route.id));
+    dispatch(spectraActions.fetchSourceSpectra(route.id));
+  }, [route.id, dispatch]);
+
+  const validateGroups = () => {
+    const formState = getValues({ nest: true });
+    return formState.groups.length >= 1;
+  };
+
+  const onSubmit = async (groupsFormData) => {
+    const selectedPhotIDs = selectedPhotRows.map((idx) => photometry[route.id][idx].id);
+    const selectedSpecIDs = selectedSpecRows.map((idx) => spectra[route.id][idx].id);
+    setIsSubmitting(true);
+    const data = {
+      groupIDs: groupsFormData.groups.map((g) => g.id),
+      photometryIDs: selectedPhotIDs,
+      spectrumIDs: selectedSpecIDs,
+    };
+    const result = await dispatch(sourceActions.shareData(data));
```

Comment:
LSP warns that this await is not being used correctly; I see we use the same pattern elsewhere.  Let's verify.